pod-operation/src/state_machine.rs

```diff
@@ -126,37 +127,33 @@ impl StateMachine {
 		//println!("Rolling INIT state");
 	}
 
-	fn _enter_init() {
-		println!("Entering INIT state");
+	/// Run the corresponding enter action for the given state
+	fn enter_state(&mut self, state: &State) {
+		let enter_action = self.enter_actions[*state];
```

Comment:
Question: In the previous version of `enter_state`, the function printed out a message if the key doesn't exist. What would happen in this version if the key doesn't exist? Or would this never happen because of the way enum maps work?